### DISCUSSION TITLE
Cloud Commander First Impressions

### DISCUSSION THREAD

**Discussion by frakman1** [coderaiser/cloudcmd#325] [Q&A]
@coderaiser 
I really enjoyed using Cloud Commander when I first started using it on my UNRAID server.
The ability to Upload and Download files is just what I was looking for.
Here are some thoughts I had and ideas to improve:

- I didn't see any documentation from the main docker hub or Github page.
 Eventually, I found the website for the app at `cloudcmd.io` here. 

![image](https://user-images.githubusercontent.com/5826484/106751505-42625480-65f7-11eb-8857-8928c4f02f74.png)

Perhaps this can be more prominantly displayed or mentioned in the README and docker hub page. Maybe a header for documentation could link there.

- The file `.cloudcmd.json` in `/root/` doesn't include the comments that are included in the homepage. These are very helpful in understanding what each option does.

File in app:

![image](https://user-images.githubusercontent.com/5826484/106751574-5dcd5f80-65f7-11eb-8308-d7942c7c3033.png)

File in website:

![image](https://user-images.githubusercontent.com/5826484/106751680-85bcc300-65f7-11eb-975e-67893aceac5c.png)

(Also, can this page be edited to make the code block wider to avoid excessive scrolling? There is a lot of wasted space on either side of the block that is not being used.)

- When opening the Config (F10) window, it would be helpful to have a tooltip for all items and not just for the pull-down menu options.

- I have no idea what the console is and what it is used for. I also don't know what commands it supports. Entering ? or help didn't yield anything. How is it different from Terminal? It seems to support a few linux-like commands. Is it supposed to be the host's terminal? If so, it doesn't have all the paths. Console usually refers to a command line way to control or view the logs for the app itself.

- I was impressed with the array of keyboard shortcuts and functionality that can be performed that way. It would be helpful if all these shortcuts are viewable in a help screen in the app itself. For example, these hot keys:

![image](https://user-images.githubusercontent.com/5826484/106753273-750d4c80-65f9-11eb-88c6-02056c333274.png)

- What do these settings mean/do?
```
    "dirStorage"            : true,     // store directory listing
    "syncConsolePath"       : false,    // do not sync console path
```
- Unchecking KeysPanel makes the bottom panel dissapear. The problem is that if I didn't remember that F10 brings up the menu, then how would I bring it back with the UI?

- The UNRAID docker template still grabs v15.5.1 when installed (instead of v15.6.0)

- Clicking the Contact button (bottom right) doesn't do anything.

- `"configAuth"            : true,     // enable auth change in config dialog` 
    This would be a good thing to have in the Config menu itself as a checkbox.

- `"columns"               : "name-size-date-owner-mode", // set visible columns` 
    Are these all the possible columns or are there more?

**Comment by coderaiser** [comment]
Thanks a lot for such a big list of suggestions :).

> I didn't see any documentation from the main docker hub or Github page.
Eventually, I found the website for the app at cloudcmd.io here.

Any contributions are welcomed, here is the file that should be changed: [README.md](https://github.com/coderaiser/cloudcmd/blob/v15.6.0/README.md).

> The file .cloudcmd.json in /root/ doesn't include the comments that are included in the homepage. These are very helpful in understanding what each option does.

I'm totally agree, but [json](https://www.json.org/json-en.html) doesn't supports comments, [json5](https://json5.org/) can be used for this purpose. But it should be tested, here is file that should be changed: [server/config](https://github.com/coderaiser/cloudcmd/blob/v15.6.0/server/config.js).

> When opening the Config (F10) window, it would be helpful to have a tooltip for all items and not just for the pull-down menu options.

Good idea :), [client/modules/config/index.js](https://github.com/coderaiser/cloudcmd/blob/v15.6.0/client/modules/config/index.js).

> (Also, can this page be edited to make the code block wider to avoid excessive scrolling? There is a lot of wasted space on either side of the block that is not being used.)

That's a good idea, here is the repo :) https://github.com/coderaiser/cloudcmd-io

> I have no idea what the console is and what it is used for. I also don't know what commands it supports. Entering ? or help didn't yield anything. How is it different from Terminal? It seems to support a few linux-like commands. Is it supposed to be the host's terminal? If so, it doesn't have all the paths. Console usually refers to a command line way to control or view the logs for the app itself.

`console` is for not-interactive programs like `touch`, `cat`, `npm` etc.
`terminal` is more powerful and can be used for interactive programs like `vim`, `mc`. It's not installed by `default` with `npm install`, but it is built-in to docker.
`terminal` can be `opend` and `closed` using `shift` + `esc`. This is done to give more keys shortcut ways, while using `terminal`, for example `vim` uses `esc` a lot.

> I was impressed with the array of keyboard shortcuts and functionality that can be performed that way. It would be helpful if all these shortcuts are viewable in a help screen in the app itself.

Just use `F1` and you will see all of them :).

> What do these settings mean/do?
    "dirStorage"            : true,     // store directory listing
    "syncConsolePath"       : false,    // do not sync console path

- `dirStorage` used to store directory listing in the `localStorage` of your web browser, this makes directory loading much faster, but file listings should be updated manually with `Ctrl` + `R`, very good for slow connections.
- `syncConsolePath` used to sync `current directory path` with current working directory in `console`.

> The UNRAID docker template still grabs v15.5.1 when installed (instead of v15.6.0)

I have no idea :), there is [v15.6.0 on dockerhub](https://hub.docker.com/r/coderaiser/cloudcmd/tags?page=1&ordering=last_updated).

> Clicking the Contact button (bottom right) doesn't do anything.
This looks like a bug in [client/modules/contact.js](https://github.com/coderaiser/cloudcmd/blob/v15.6.0/client/modules/contact.js), need help with https://www.olark.com/api.

> Unchecking KeysPanel makes the bottom panel dissapear. The problem is that if I didn't remember that F10 brings up the menu, then how would I bring it back with the UI?

Just use right mouse button.

> "configAuth" : true, // enable auth change in config dialog
This would be a good thing to have in the Config menu itself as a checkbox.

Don't know, this option not so often enabled/disabled.

> "columns" : "name-size-date-owner-mode", // set visible columns
Are these all the possible columns or are there more?

Yes, but pull requests are welcomed :)

**Reply by frakman1** [reply]
Thank you for addressing all these issues. Regarding this:

>  Unchecking KeysPanel makes the bottom panel dissapear. The problem is that if I didn't remember that F10 brings up the menu, then how would I bring it back with the UI?

>  Just use right mouse button.

Right mouse button just brings up this menu:

![image](https://user-images.githubusercontent.com/5826484/106817933-8f6d1780-6645-11eb-93ab-13ac2d62249d.png)

I was meaning this:

![image](https://user-images.githubusercontent.com/5826484/106817986-a0b62400-6645-11eb-86eb-f7aafb723567.png)

**Reply by frakman1** [reply]
I took a crack at a [pull request](https://github.com/coderaiser/cloudcmd/pull/326). Let me know if that looks OK.
